Null body generates an NPE in find* operations

This code:
```
def foo = body.find { something }
```
can lead to:
```
Caused by: java.lang.NullPointerException
    at java.util.Collections$UnmodifiableCollection.<init>(Collections.java:1026)
    at java.util.Collections$UnmodifiableList.<init>(Collections.java:1302)
    at java.util.Collections.unmodifiableList(Collections.java:1287)
```

--- a/webtau-rest-groovy/src/test/groovy/com/twosigma/webtau/http/HttpTest.groovy
+++ b/webtau-rest-groovy/src/test/groovy/com/twosigma/webtau/http/HttpTest.groovy
@@ -175,6 +175,15 @@ class HttpTest {
     }
 
     @Test
+    void "groovy find on body that is not a list"() {
+        def found = http.get("/end-point") {
+            return body.find { it > 1 }
+        }
+
+        assert found == null
+    }
+
+    @Test
     void "groovy findAll on list"() {
         def found = http.get("/end-point") {
             return list.findAll { it > 1 }
@@ -182,6 +191,15 @@ class HttpTest {
 
         assert found == [2, 3]
         assert found[0].getClass() == Integer
+    }
+
+    @Test
+    void "groovy findAll on body that is not a list"() {
+        def found = http.get("/end-point") {
+            return body.findAll { it > 1 }
+        }
+
+        assert found == []
     }
 
     @Test
@@ -201,12 +219,21 @@ class HttpTest {
 
     @Test
     void "groovy transform list"() {
-        def found = http.get("/end-point") {
+        def transformed = http.get("/end-point") {
             return list.collect { "world#${it}" }
         }
 
-        assert found == ['world#1', 'world#2', 'world#3']
-        assert found[0] instanceof GString
+        assert transformed == ['world#1', 'world#2', 'world#3']
+        assert transformed[0] instanceof GString
+    }
+
+    @Test
+    void "groovy transform on body that is not a list"() {
+        def transformed = http.get("/end-point") {
+            return body.collect { "world#${it}" }
+        }
+
+        assert transformed == []
     }
 
     @Test

--- a/webtau-rest/src/main/java/com/twosigma/webtau/http/datanode/StructuredDataNode.java
+++ b/webtau-rest/src/main/java/com/twosigma/webtau/http/datanode/StructuredDataNode.java
@@ -16,12 +16,11 @@
 
 package com.twosigma.webtau.http.datanode;
 
+import com.twosigma.webtau.data.traceable.TraceableValue;
+
 import java.util.Collections;
 import java.util.List;
 import java.util.Map;
-import java.util.stream.Collectors;
-
-import com.twosigma.webtau.data.traceable.TraceableValue;
 
 import static java.util.stream.Collectors.joining;
 
@@ -84,7 +83,9 @@ public class StructuredDataNode implements DataNode {
 
     @Override
     public List<DataNode> elements() {
-        return Collections.unmodifiableList(values);
+        return values == null ?
+                Collections.emptyList() :
+                Collections.unmodifiableList(values);
     }
 
     @Override

